security/secret-management.md

```diff
@@ -0,0 +1,112 @@
+
+
+### Report
+1. **Document the Incident**: Record details of the leak in GitGuardian, including the type of secret, the steps taken 
+   to remediate it, and any lessons learned. 
+2. **Implement Preventive Measures**: Update security practices, tools, and workflows to prevent similar incidents in 
+   the future. This may include enhancing code review practices, improving secret management practices, or increasing 
+   training for developers.
+
+## Best Practices for Secret Management
+
+   **Environment Variables**: Store secrets in environment variables instead of hardcoding them in the source code or
+   properties files
+   **Secret Management Tools**: Utilize dedicated secret management tools like AWS Secrets Manager to manage and store 
+   secrets securely.
+   **Principle of least privilege**: Ensure that only necessary services and users have access to secrets.
+
+
+## Responsibility for Secret Management
+
+To ensure the security and integrity of our secrets, the Web Operations team will be responsible for the storage and 
+management of all production and development secrets. This centralized approach helps maintain control over access and 
+ensures that only authorized personnel have the ability to manage and access sensitive information.
+
+### Responsibilities of the Operations Team:
+
+- **Secure Storage**: Manage and store all production and test secrets using secure and approved secret management tools 
+  and practices.
+- **Issue Secrets**: Create new secrets when access to sensible systems is needed. This responsibility can be delegated
+  to helpdesk for some systems.
+- **Access Control**: Implement strict access controls and policies to ensure that only authorized team members and 
+  systems can access production secrets. Regularly review and update access permissions. Test secrets should be 
+  evaluated depending on the secret itself as some of them might need to be carefully handled to prevent a breach of 
+  contract
+- **Audit**: Conduct regular audits of secret access and usage and revoke access to systems if it is no longer needed
+
+### Developer Responsibilities
+
+- **Handle secrets**: All developers are responsible for following the best practices outlined in this policy to ensure 
+  secrets are handled securely. They cannot share _secrets_ or create new ones based on existing ones unless it is 
+  approved by the Operations team.
+- **Do Not Hardcode Secrets**: Avoid hardcoding secrets directly in the codebase. Use environment variables, 
+  configuration files, or secure vaults to manage secrets.
+- **Report Issues**: Immediately report any suspected leaks or vulnerabilities related to secrets to the Web Operations 
+  team. Prompt reporting helps mitigate risks and prevent potential security incidents.
+
+## Policy Compliance
+Compliance with this policy is mandatory. Failure to comply with this policy will result in stricter processes
```

Comment:
Hello @BrianGibbVS,

The purpose of that sentence is to warn the team that this process will work if we collaborate to make it work. If this doesn't work, we need to evaluate how, as a team, we create stricter rules, so the process does not fall through the cracks. 

That's why at this point it might be better to keep this as a warning, but we can take it out if you think that doesn't work

Reply:
@jcalcines - I would probabaly remove this sentence... I think it raises more questions than provides answers.

If we just say that the policy is mandatory and finish there, that shoud be enough to state the importance of it.

Reply:
I have removed the last sentence, concluding just with "The compliance of this policy is mandatory"